infer/types/asset.go

```diff
@@ -0,0 +1,29 @@
+// Copyright 2022, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package types
+
+import "github.com/pulumi/pulumi/sdk/v3/go/common/resource"
+
+const AssetSignature = "a9e28acb8ab501f883219e7c9f624fb6"
+const ArchiveSignature = "195f3948f6769324d4661e1e245f3a4d"
```

Comment:
Let's try to add a docstring to each exposed part of the API.

```suggestion
const (
    // The signature for the asset part of [AssetOrArchive]
    AssetSignature = "a9e28acb8ab501f883219e7c9f624fb6"
    // The signature for the archive part of [AssetOrArchive]
    ArchiveSignature = "195f3948f6769324d4661e1e245f3a4d"
)
```

Even better, could we move these to `ende` and keep them private. I don't want them to show up in autocomplete since users should never need to use them.

Reply:
Ah, good call. In contrast to the type, the constants are only used in ende.